sequential/backend/build_mdd.cpp

```diff
@@ -98,16 +98,18 @@ bool Check_gap(int i, int strp, int endp, vector<vector<vector<int> > >* attrs,
 vector<int>* lgap, vector<int>* ugap){	//checks upper and lower gap constraints imposed on any attribute value
 
 	for (int att = 0; att < (*lgap).size(); att++){
-		if ((*lgap)[att] == 0)
-			continue;
-		if (abs((*attrs)[(*lgapi)[att]].at(i).at(endp - 1) - (*attrs)[(*lgapi)[att]].at(i).at(strp - 1)) < (*lgap)[att])
+// Allow lower bound of gap constraint to be 0 if not work with abs values
```

Comment:
This is the fix that Amin did in the original/earlier repository, correct? 

My suggestions is; go that repo and look at the latest commits. It should be easy to find the commit for this fix at the backend. Then, you can compare & contrast the solutions. 

I vaguely remember looking into it at the time (~2 years ago) and it was a simple/one-liner change like this. 

It should be easy to verify.

Reply:
I see his original repo uses abs value at this line though (see the [lines](https://github.com/aminhn/MPP/blob/master/src/build_mdd.cpp#L81)), which is the latest commit. 

In an earlier commit, there is a fix at another line ([line 46](https://github.com/aminhn/MPP/commit/009157d2a547b5481aa03bce169f755b3c2d3f4a#diff-b6d92d609a9a4838760571e76d844b5908d71648ecd54860becb1979c595037bR46)), which was to apply abs instead of removing it. So I tried to do the same fix as Amin did, which didn't give me expected results as I have negative upper/lower bounds and run it on paper though.

I am thinking that maybe I can leave an inline comment (shown as an issue) to the original repo and Amin can provide his thoughts there?